### ISSUE TITLE
Feature request: Using SIMD for byte search

### ISSUE THREAD

**Issue by He-Pin** [apache/pekko#1264]
Motivation:
When reading the code in https://github.com/apache/pekko/pull/1247 , I recall that we can improve performance with SIDM.
related blog:  https://richardstartin.github.io/posts/finding-bytes

and for reference in Netty: https://github.com/netty/netty/pull/10737

**Comment by samueleresca** [comment]
@He-Pin Thanks for sharing the blog post 😄. Interesting stuff

**Comment by He-Pin** [comment]
> @He-Pin Thanks for sharing the blog post 😄. Interesting stuff

Yes, with it, we can test 8 byte in a single loop，I think this is supported with SSE, AVX is 128 bits and AVX2 is 256 bits

**Comment by laglangyue** [comment]
so cool

**Comment by JD557** [comment]
> Would you like to take a look at this too @JD557

Unfortunately, I don't think I'll have capacity to look into this in the near future. Also, my knowledge of SIMD is pretty limited.

Having said that:

> But without the vector api that what we can do now.

I think there's value in first trying something that is *not* using the Vector API (like the code suggested above). That way Pekko can keep supporting old Java versions, as well as being easier to cross compile to JS and Native.

Hopefully that would hit the autovectorizaton logic in some of those platforms, but I haven't tested that 😅

**Comment by pjfanning** [comment]
#2148 was merged
